### PR TITLE
 Generate flex yaml file in temp folder when testing non-yaml flex flow 

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

1. --ui will generate .promptflow in temp folder since will specify output folder as flow yaml parent folder
image
image

2. cli test will generate .promptflow in code folder when not specify output folder in flow test because submitter use code path as default output path
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py [hunk 1]
```diff
@@ -975,6 +974,30 @@ def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code:
...
+    logger.info("Create temporary entry for flex flow.")
+    entry, code = resolve_entry_and_code(entry, code)
+    flow_dag = {"entry": entry, "code": str(code.resolve())}
+    temp_dir = tempfile.mkdtemp()
+    update_signatures(code=code, data=flow_dag)
+    flow_yaml_path = dump_flow_dag_according_to_content(flow_dag=flow_dag, flow_path=Path(temp_dir))
+    return flow_yaml_path
+
+
+def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code: Path = None):
```
Comment (D-W-) [inline src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py:998]: May direct change whole `create_temp_flex_flow_xxx`'s behavior to genenerate in tmp folder.

File: src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py [hunk 2]
```diff
@@ -975,6 +974,30 @@ def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code:
...
+    """
+
+    from promptflow._sdk._utilities.signature_utils import update_signatures
+    from promptflow._utils.flow_utils import dump_flow_dag_according_to_content
+
+    logger.info("Create temporary entry for flex flow.")
+    entry, code = resolve_entry_and_code(entry, code)
+    flow_dag = {"entry": entry, "code": str(code.resolve())}
+    temp_dir = tempfile.mkdtemp()
+    update_signatures(code=code, data=flow_dag)
```
Comment (D-W-) [inline src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py:993]: Will this be used by UI?

Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py]: removed

File: src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py (hunk 1 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py]: updated

File: src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py [hunk 3]
```diff
@@ -975,6 +975,25 @@ def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code:
...
+
+def create_flex_flow_yaml_in_target(entry: Union[str, PathLike, Callable], target_dir: str, code: Path = None):
+    """
+    Generate a flex flow yaml in target folder. The code field in the yaml points to the original flex yaml flow folder.
+    """
+    from promptflow._utils.flow_utils import dump_flow_dag_according_to_content
+
+    logger.info("Create temporary entry for flex flow.")
+    entry, code = resolve_entry_and_code(entry, code)
+    flow_dag = {"entry": entry, "code": str(code.resolve())}
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py:989]: if code is same as target_dir, we can log ".' here

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py (hunk 3 above)
Comment (YingChen1996) [reply src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py]: updated

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**